src/pages/files.js

```diff
@@ -88,112 +130,207 @@ export default class Files extends React.Component {
     });
   };
 
+  handleBackMyStack = () => {
+    this.setState({
+      showModal: true,
+    });
+  };
+
+  handleContinueBacMyStack = () => {
+    // Get the key url of the file
+    const { savedFileUrl } = this.state;
+    const { openCollectiveRedirectUrl } = this.props;
+    const { hostname, protocol } = window.location;
+    const uuid = savedFileUrl.Key.split('/')[0];
+    const jsonUrl = `${protocol}//${hostname}/${uuid}/file/backing.json`;
+    const data = JSON.stringify({
+      jsonUrl,
+    });
+    const contributionUrl = `${openCollectiveRedirectUrl}?data={${data}}`;
```

Comment:
This stills looks not correct.

- Looks like the JSON will have duplicate braces `{${data}}`
- URL query parameters (here `data`) needs to be properly handled, for example with the URL API: https://developer.mozilla.org/en-US/docs/Web/API/URL

Reply:
Noticed already, on it